pyclam/manifold.py

```diff
@@ -133,13 +135,15 @@ def argsamples(self) -> Vector:
             else:
                 n = int(np.sqrt(len(self)))
                 indices = list(np.random.choice(self.argpoints, n, replace=False))
+                indices = [int(i) for i in indices]
 
             # Handle Duplicates.
-            if pdist(self.manifold.data[indices], self.metric).max(initial=0.) == 0.:
+            if self.distance(x1=indices, x2=indices).max(initial=0.) == 0.:
                 indices = np.unique(self.manifold.data[self.argpoints], return_index=True, axis=0)[1]
                 indices = [self.argpoints[i] for i in indices][:n]
 
             # Cache it.
+            indices = [int(i) for i in indices]
```

Comment:
Why int here, they should be ints already, if they aren't we fucked up elsewhere

Reply:
Line 143 makes it a list of numpy.int64